From: Arnaud Ferraris <arnaud.ferraris@collabora.com>
Date: Wed, 3 Aug 2022 22:53:21 +0200
Subject: meson.build: ensure libsqueekboard is built first

This is required for our ppc64el workaround to build.
---
 src/meson.build   | 27 +++++++++++++--------------
 tools/meson.build |  2 +-
 2 files changed, 14 insertions(+), 15 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index af33418..9db99a4 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -55,6 +55,17 @@ deps = [
 #  dependency('libxklavier'), # FIXME remove
 ]
 
+libsqueekboard = static_library('squeekboard',
+  sources,
+  include_directories: [include_directories('..'), include_directories('../eek')],
+  dependencies: deps,
+  c_args: [
+    '-DTHEMESDIR="' + pkgdatadir + '/themes"',
+    '-DKEYBOARDSDIR="' + pkgdatadir + '/keyboards"',
+    '-DEEKBOARD_COMPILATION=1',
+    '-DEEK_COMPILATION=1'],
+)
+
 rslibs = custom_target(
     'rslibs',
     build_by_default: true,
@@ -63,7 +74,7 @@ rslibs = custom_target(
     install: false,
     console: true,
     command: [cargo_build] + ['@OUTPUT@', '--lib'] + cargo_build_flags,
-    depends: cargo_toml,
+    depends: [cargo_toml, libsqueekboard],
 )
 
 build_rstests = custom_target(
@@ -91,21 +102,9 @@ test(
     depends: [build_rstests, cargo_toml],
 )
 
-libsqueekboard = static_library('libsqueekboard',
-  sources,
-  link_with: [rslibs],
-  include_directories: [include_directories('..'), include_directories('../eek')],
-  dependencies: deps,
-  c_args: [
-    '-DTHEMESDIR="' + pkgdatadir + '/themes"',
-    '-DKEYBOARDSDIR="' + pkgdatadir + '/keyboards"',
-    '-DEEKBOARD_COMPILATION=1',
-    '-DEEK_COMPILATION=1'],
-)
-
 squeekboard = executable('squeekboard',
   'server-main.c',
-  link_with: libsqueekboard,
+  link_with: [libsqueekboard, rslibs],
   include_directories: [include_directories('..'), include_directories('../eek')],
   dependencies: deps,
   install: true,
diff --git a/tools/meson.build b/tools/meson.build
index 75a7e17..27f6ae1 100644
--- a/tools/meson.build
+++ b/tools/meson.build
@@ -16,5 +16,5 @@ test_layout = custom_target('squeekboard-test-layout',
         + cargo_build_flags,
     install: true,
     install_dir: bindir,
-    depends: cargo_toml,
+    depends: rslibs,
 )
